targets/unity-v2/templates/PlayFab_Events_InstanceAPI.cs.ejs

```diff
@@ -0,0 +1,124 @@
+        }
+
+        public PlayFab<%- api.name %>InstanceAPI(PlayFabApiSettings settings, PlayFabAuthenticationContext context)
+        {
+            if (context == null)
+                throw new PlayFabException(PlayFabExceptionCode.AuthContextRequired, "Context cannot be null, create a PlayFabAuthenticationContext for each player in advance, or call <PlayFabClientInstanceAPI>.GetAuthenticationContext()");
+            apiSettings = settings;
+            authenticationContext = context;
+        }
+<% } %><% if (hasClientOptions) { %>
+        /// <summary>
+        /// Verify client login.
+        /// </summary>
+        public bool IsClientLoggedIn()
+        {
+            return authenticationContext == null ? false : authenticationContext.IsClientLoggedIn();
+        }
+<% } %><% if (hasEntityTokenOptions) { %>
+        /// <summary>
+        /// Verify entity login.
+        /// </summary>
+        public bool IsEntityLoggedIn()
+        {
+            return authenticationContext == null ? false : authenticationContext.IsEntityLoggedIn();
+        }
+<% } %>
+        /// <summary>
+        /// Clear the Client SessionToken which allows this Client to call API calls requiring login.
+        /// A new/fresh login will be required after calling this.
+        /// </summary>
+        public void ForgetAllCredentials()
+        {
+            if (authenticationContext != null)
+            {
+                authenticationContext.ForgetAllCredentials();
+            }
+        }
+<% for(var i in api.calls) { var apiCall = api.calls[i]; %>
+<%- generateApiSummary("        ", apiCall, "summary") %><%- getDeprecationAttribute("        ", apiCall)
+%>        public void <%- apiCall.name %>(<%- apiCall.request %> request, Action<<%- apiCall.result %>> resultCallback, Action<PlayFabError> errorCallback, object customData = null, Dictionary<string, string> extraHeaders = null)
+        {
+            var context = (request == null ? null : request.AuthenticationContext) ?? authenticationContext;
+            var callSettings = apiSettings ?? PlayFabSettings.staticSettings;
+<%- getRequestActions("            ", apiCall) %>            PlayFabHttp.MakeApiCall("<%- apiCall.url %>", request, <%- getAuthParams(apiCall) %>, resultCallback, errorCallback, customData, extraHeaders, context, callSettings, this);
+        }<%- getCustomApiFunction("        ", api, apiCall, true) %>
+<% } %> <% if (apiCall.name == "WriteTelemetryEvents") 
+{ %>       /// <summary>
+        /// Write batches of entity based events to as Telemetry events (bypass PlayStream) using a Telemetry Key. The namespace must be 'custom' or start
+        /// with 'custom.'
+        /// </summary>
+        public void <%- apiCall.name %>WithTelemetryKey(<%- apiCall.request %> request, Action<<%- apiCall.result %>> resultCallback, Action<PlayFabError> errorCallback, object customData = null, Dictionary<string, string> extraHeaders = null)
```

Comment:
I basically added a new API called _WriteTelemetryEventsAsyncWithTelemetryKey_ as opposed with the C# SDK where I used an API overload. We could probably also use API overload but it would require a change on the PlayFabHttp template which would represent a greater change.